### PR TITLE
Improve page titles for better SEO

### PR DESCRIPTION
**Pull request by Taha-Hassan-Git**
### SEO Title Changes:

- Lesson overview pages: [lesson name] KS[X] | Y[X] [subject] Lesson Resources
- Lesson listing pages: [unit name] KS[X] | Y[X] [subject] Lesson Resources
- Unit listing pages: Free KS[X] [subject] teaching resources | [Year groups as Y1, 2, & 3]

- Year group object refactor: Split the year field into:
   - yearSlug: The URL-friendly slug (e.g., "year-10")
   - year: The numeric/abbreviated year (e.g., "10", "R" for reception)

- New utility function: Added getYearGroupSEOString() to format year groups for SEO titles (e.g., "Y1", "Y1 & 2", "Y1, 2, & 3") along with tests
- Schema updates: Updated curriculum API schemas to include the new year field
- Query updates: Updated curriculum API queries to extract and return the new year field from database results
- Component updates: Updated all components using year group data to use the correct field names (yearSlug for routing, year for display)
- Test updates: Updated fixtures and tests to reflect the new data structure
- Filter logic updates: Updated unit filtering logic to use yearSlug instead of year for URL routing

## Issue(s)

Fixes [#1417](https://www.notion.so/oaknationalacademy/I-want-more-SEO-friendly-page-titles-1f926cc4e1b1806fb61fc5885c465a12)

## How to test

1. Go to a lesson overview page, make sure it follows the pattern ```[lesson name] KS[X] | Y[X] [subject] Lesson Resources | Oak National Academy```
2.  Go to a lesson listing page, make sure it follows the pattern ```Unit: [unit name] KS[X] | Y[X] [subject] Lesson Resources | Oak National Academy```
3. Go to a unit listing page make sure it follows the pattern ```Free KS[x] [subject] teaching resources | [Year groups in KS presented as Y x, y & z] | Page 1 of 5 | Oak National Academy```

I'm not so familiar with our data, so please be aware of any edge cases, for example are year groups always present? is there ever pagination on the lesson overview page?

## Screenshots

Kapture 2025-06-04 at 09 37 02

## Checklist

- [x] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [x] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx [hunk 1]
```diff
@@ -219,12 +220,11 @@ const LessonListPage: NextPage<LessonListingPageProps> = ({
       subjectSlug,
     },
   );
-
   return (
     <AppLayout
       seoProps={{
         ...getSeoProps({
-          title: `Unit: ${unitTitle} | ${keyStageSlug.toUpperCase()} ${subjectTitle}${paginationTitle}`,
+          title: `Unit: ${unitTitle} ${keyStageSlug.toUpperCase()} | Y${yearTitle.slice(5)} ${subjectTitle} Lesson Resources ${paginationTitle}`,
```
Comment (left23) [inline src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx:227]: There's a discrepancy on the ticket between what is in its description and what is in the acceptance criteria...
I _think_ the word "Unit: " should be omitted here, based on the nature of the changes being requested. This would align with what's in the AC on the ticket (not the description), and IMO is more informative to the end user when they see it in their browser tabs.

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx [hunk 2]
```diff
@@ -305,12 +305,12 @@ describe("pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[le
       expect(seo).toEqual(
         expect.objectContaining({
           title:
-            "Lesson: Adverbial complex sentences | AQA | KS2 English | NEXT_PUBLIC_SEO_APP_NAME",
+            "Adverbial complex sentences | AQA KS2 | Y3 English Lesson Resources | NEXT_PUBLIC_SEO_APP_NAME",
```
Comment (left23) [inline src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx:308]: As above, I think we can omit the pipe between the lesson title and the exam board so it would be: "Adverbial complex sentences AQA KS2 | Y3 ... "

**Review by left23 — CHANGES_REQUESTED** [review]
@Taha-Hassan-Git Thanks, I've made some comments on the test data, to demonstrate the inconsistencies in the format when there is a tier or exam board or both. I'm not sure what the desired end result is because the ticket doesn't make it clear in these cases either.
I think it comes down to either the unit and lesson titles are followed by a pipe all the time or never. At the moment, it seems a bit of both depending on whether the titles are followed by exam board, tier, both, or just the key stage. I hope that's enough to go on the make it more consistent. No preference from me as to which. Maybe @njblore can advise.

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/unitListing.test.tsx [hunk 3]
```diff
@@ -119,11 +119,11 @@ describe("pages/programmes/[programmeSlug]/units", () => {
         ...mockSeoResult,
         ogSiteName: "NEXT_PUBLIC_SEO_APP_NAME",
         title:
-          "Free KS4 Computing teaching resources | NEXT_PUBLIC_SEO_APP_NAME",
+          "Free KS4 Computing teaching resources | Y 10  | NEXT_PUBLIC_SEO_APP_NAME",
```
[the remark this answers is not in the document]
Comment (left23) [reply src/__tests__/pages/teachers/programmes/[programmeSlug]/units/unitListing.test.tsx:122]: Now I'm not sure about the space needing to be there or not... See https://deploy-preview-3464--oak-web-application.netlify.thenational.academy/teachers/programmes/combined-science-secondary-ks4-foundation-aqa/units
Maybe it's nice to have additional test data for where there are multiple years mentioned `Free KS4 Combined science teaching resources | Y 10 &amp; 11  | Page 1 of 3 | Oak National Academy`.
The space after the Y makes more sense when there are multiple years but maybe it's still ok as "Y10 & 11" ?

File: src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx (hunk 1 above)
Comment (Taha-Hassan-Git) [reply src/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons.tsx:227]: That makes sense, I agree.

File: src/pages/teachers/programmes/[programmeSlug]/units.tsx [hunk 4]
```diff
@@ -124,9 +124,26 @@ const UnitListingPage: NextPage<UnitListingPageProps> = ({
   const learningThemesId = useId();
   const learningThemesFilterId = useId();
 
+  // formats the year groups as e.g: Y 6, 7, 8 & 9
+  const yearGroupSEOString = (() => {
+    if (yearGroups.length === 0) return "";
+    if (yearGroups.length === 1) {
+      return `Y ${yearGroups[0]!.yearTitle.slice(5)}`;
```
[the remark this answers is not in the document]
Comment (Taha-Hassan-Git) [reply src/pages/teachers/programmes/[programmeSlug]/units.tsx]: I've looked at the schema and found that it'll sometimes have "All years" as well. Rather than amend the query, I've written a function that uses the yearTitle to format the string. I've extracted it to a pages-helpers and added a test file. It's maybe a bit overcomplicated, but gets the job done. Keen to hear what you think. Tests were useful while developing but might be a bit overkill just for preventing regression, so I could fix cut some of them. 

I made some assumptions about the data that I'm hopeful you can confirm:

1. Year groups are always in sequential order and do not skip years
2. "All years" will only appear as a single item in the array
3. "Reception" will only appear as the first year in an array

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx [hunk 5]
```diff
@@ -330,11 +330,11 @@ describe("pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[le
       expect(seo).toEqual(
         expect.objectContaining({
           title:
-            "Lesson: Adverbial complex sentences | Higher | AQA | KS2 English | NEXT_PUBLIC_SEO_APP_NAME",
+            "Adverbial complex sentences Higher | AQA KS2 | Y3 English Lesson Resources | NEXT_PUBLIC_SEO_APP_NAME",
```
Comment (left23) [inline src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx:333]: I think the pipe between the tier and the exam board here should be omitted, to be consistent with the treatment where there is only a tier (line 293)
So it would read: `Adverbial complex sentences Higher AQA KS2 | Y3 English ... `

For example, this https://deploy-preview-3464--oak-web-application.netlify.thenational.academy/teachers/programmes/combined-science-secondary-ks4-higher-edexcel/units/atomic-structure-and-the-periodic-table/lessons/atomic-structure-negligible-electron-mass would read `Atomic structure (negligible electron mass) Higher Edexcel KS4 | Y10 Combined science Lesson Resources | Oak National Academy`

File: src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOverview.schema.ts [hunk 6]
```diff
@@ -61,6 +61,7 @@ export const lessonOverviewSchema = baseLessonOverviewSchema.extend({
   subjectTitle: z.string(),
   subjectParent: z.string().nullish(),
   yearTitle: z.string().nullable().optional(),
+  year: z.string().nullable().optional(),
```
Comment (njblore) [inline src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOverview.schema.ts:64]: there's a shortcut for this `.nullish()`, I'm not sure why it hasn't been used for some of the other nullable+optional fields !

File: src/pages-helpers/teacher/year-group-seo-string/get-year-grp-seo-string.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { YearGroups } from "../../../node-lib/curriculum-api-2023/queries/unitListing/unitListing.schema";
```
Comment (njblore) [inline src/pages-helpers/teacher/year-group-seo-string/get-year-grp-seo-string.ts:1]: `"@/node-lib/curriculum-api-2023/queries/unitListing/unitListing.schema"`

**Review by njblore — APPROVED** [review]
looking good!
